src/joint.dia.element.js

```diff
@@ -694,8 +694,6 @@ joint.dia.ElementView = joint.dia.CellView.extend({
 
     pointerdown: function(evt, x, y) {
 
-        this.model.trigger('batch:start');
-
         // target is a valid magnet start linking
         if (evt.target.getAttribute('magnet') && this.paper.options.validateMagnet.call(this.paper, this, evt.target)) {
 
```

Comment:
James I think you've kind of done what you described. I can not see any double calls anymore.
- A batch named `pointer` can be triggered by a link or element. That's why it's handled on the cellView.
- A batch names `add-link` can be triggered by an element only. That's why it's handled on the ElementView.
- Note also there is a bunch of other `batch:start`/`batch:stop` triggered on the model (`to-front`, `to-back`, `embed`, `unembed`, `resize`, `remove`, `clear`). 

I'm very happy with what you've done here. I think it's ready now. Do you mind run `git rebase -i HEAD~10` and squash commits before we merge it? Cheers.